agent_plugin_builder/vendor_dirs.py

```diff
@@ -59,70 +66,140 @@ def check_if_common_vendor_dir_possible(build_dir: Path, uid, gid) -> bool:
     windows_packages_path.unlink()
 
     response = all_equal([linux_packages, windows_packages])
+    if response:
+        logger.info("Common vendor directory is possible")
+    else:
+        logger.info("Common vendor directory is not possible")
 
     return response
 
 
-def generate_common_vendor_dir(build_dir: Path, uid, gid):
+def generate_common_vendor_dir(build_dir: Path, uid: int, gid: int):
+    """
+    Generate a common vendor directory by installing the requirements in a Linux container.
+
+    :param build_dir: Path to the build directory.
+    :param uid: User ID to set on the vendor directory.
+    :param gid: Group ID to set on the vendor directory.
+    """
     client = docker.from_env()
     commands = LINUX_IMAGE_PYENV_INIT_COMMANDS + [
         "cd /plugin && pip install -r requirements.txt -t src/vendor",
-        f"chown -R {uid}:{gid} /plugin/src/vendor",
+        "chmod 755 /plugin/src/vendor",
     ]
 
     full_command = "/bin/bash -c '" + " && ".join(commands) + "'"
 
-    client.containers.run(
+    linux_container = client.containers.run(
         AGENT_PLUGIN_IMAGE,
         command=full_command,
         volumes={str(build_dir): {"bind": "/plugin", "mode": "rw"}},
         remove=True,
     )
+    _log_container_output(linux_container, "Common vendor directory, ")
+
 
+def generate_vendor_dirs(build_dir: Path, operating_system: OperatingSystem, uid: int, gid: int):
+    """
+    Generate the vendor directories for the plugin.
 
-def generate_vendor_dirs(build_dir: Path, operating_system: OperatingSystem, uid, gid):
+    :param build_dir: Path to the build directory.
+    :param operating_system: Operating system to generate the vendor directories for.
+    :param uid: User ID to set on the vendor directory.
+    :param gid: Group ID to set on the vendor directory.
+    """
     if operating_system == OperatingSystem.LINUX:
         generate_linux_vendor_dir(build_dir, uid, gid)
     elif operating_system == OperatingSystem.WINDOWS:
         generate_windows_vendor_dir(build_dir, uid, gid)
 
 
-def generate_linux_vendor_dir(build_dir: Path, uid, gid):
+def generate_linux_vendor_dir(build_dir: Path, uid: int, gid: int):
+    """
+    Generate the Linux vendor directory by installing the requirements in a Linux container.
+
+    :param build_dir: Path to the build directory.
+    :param uid: User ID to set on the vendor directory.
+    :param gid: Group ID to set on the vendor directory.
+    """
     client = docker.from_env()
     commands = LINUX_IMAGE_PYENV_INIT_COMMANDS + [
         "cd /plugin && pip install -r requirements.txt -t src/vendor-linux",
-        f"chown -R {uid}:{gid} /plugin/src/vendor-linux",
+        "chmod 755 /plugin/src/vendor-linux",
     ]
 
     full_command = "/bin/bash -c '" + " && ".join(commands) + "'"
```

Comment:
We should be very careful about concatenating shell commands so that we avoid any risk of OS command injection. Ideally we wouldn't be dynamically building shell commands.